dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -40,43 +39,100 @@ export function Table(options: TableData) {
   };
 
   const columns = [
+    {
+      field: 'type',
+      name: '',
+      width: '3%',
+      render: (item) => {
+        if (item == 'Visualization') {
+          return (
+            <div>
+              <EuiLoadingChart size="m" />
+            </div>
+          );
+        } else {
+          return (
+            <div>
+              <EuiIcon type="discoverApp" size="m" />
+            </div>
+          );
+        }
+      },
+    },
     {
       field: 'data',
       name: 'Name',
-      render: (item)=>{return <EuiLink onClick={() =>
-      {options.savedQuerySearch(item.query, [item.date_start, item.date_end], item.timestamp, item.fields)}}>
-        {item.name}
-      </EuiLink>},
+      align: 'left',
+      render: (item) => {
+        return (
+          <EuiLink
+            onClick={() => {
+              options.savedQuerySearch(
+                item.query,
+                [item.date_start, item.date_end],
+                item.timestamp,
+                item.fields
+              );
+            }}
+          >
+            {item.name}
+          </EuiLink>
+        );
+      },
     },
     {
-      field: 'description',
-      name: 'Description',
+      field: 'type',
+      name: 'Type',
+      align: 'right',
     },
   ];
 
-
-
+  let objectName = '';
   const queries = options.savedHistory.map((h) => {
     const savedObject = h.hasOwnProperty('savedVisualization')
       ? h.savedVisualization
       : h.savedQuery;
+    const object = h.hasOwnProperty('savedVisualization');
```

Comment:
Maybe a more descriptive name like isSavedVisualization?

Reply:
I'll change it